examples/silentpayments.c

```diff
@@ -0,0 +1,432 @@
+            assert(ret);
+
+            /* Scan the transaction */
+            n_found_outputs = 0;
+            ret = secp256k1_silentpayments_recipient_scan_outputs(ctx,
+                found_output_ptrs, &n_found_outputs,
+                tx_output_ptrs, N_TX_OUTPUTS,
+                bob_scan_key,
+                &public_data,
+                &spend_pubkey,
+                label_lookup, &labels_cache /* NULL, NULL if scanning without labels */
+            );
+            assert(n_found_outputs == 1);
+            printf("\n");
+            printf("Bob found the following outputs: \n");
+            for (i = 0; i < n_found_outputs; i++) {
+                printf("    ");
+                secp256k1_xonly_pubkey_serialize(ctx, xonly_print, &found_outputs[i].output);
+                print_hex(xonly_print, sizeof(xonly_print));
+            }
+        }
+        {
+            /*** Scanning as a light client (Carol) ***
+             *
+             * Being a light client, Carol likely does not have access to the transaction outputs. This
+             * means she will need to first generate an output, check if it exists in the UTXO set (e.g.
+             * BIP158 or some other means of querying) and only proceed to check the next output (by
+             * incrementing `k`) if the first output exists.
+             *
+             * Additionally, Carol likely does not have access to the transaction inputs and prevout information,
+             * so she uses the `public_data` object creatd by Bob's full node earlier. This serialized `public_data` object
+             * contains everything she needs for generating the shared secret, i.e., `input_hash * A_sum`.
+             *
+             * In practice, Carol wouldn't know the number of outputs ahead of time but we are cheating here
+             * to keep the example simple.
+             */
+            unsigned char ser_found_outputs[2][32];
+            unsigned char shared_secret[33];
+            secp256k1_pubkey spend_pubkey;
+            secp256k1_silentpayments_public_data public_data;
+            size_t n_found_outputs;
+
+            /* Load Carol's spend public key */
+            ret = secp256k1_ec_pubkey_parse(ctx, &spend_pubkey, carol_address[1], 33);
+            assert(ret);
+
+            /* Scan one output at a time, using the serialized `public_data` created by Bob's full node */
+            ret = secp256k1_silentpayments_recipient_public_data_parse(ctx, &public_data, light_client_data33);
+            assert(ret);
+            ret = secp256k1_silentpayments_recipient_create_shared_secret(ctx,
+                shared_secret,
+                carol_scan_key,
+                &public_data
+            );
+            assert(ret);
+            n_found_outputs = 0;
+            {
+                int found = 0;
+                size_t k = 0;
+                secp256k1_xonly_pubkey potential_output;
+
+                while(1) {
+
```

Comment:
nit: add space after `while` and remove the empty line

Reply:
Missed this but will address in the next round of review

Reply:
Fixed.